src/Reflection/ReflectionParameter.php

```diff
@@ -497,8 +496,26 @@ private function getClassName(): ?string
         }
 
         $type = $this->getType();
-        assert($type instanceof ReflectionType);
```

Comment:
Is the `hasType` call above even needed, at this point? I suppose `getType()` would suffice by now, and if it's `null`, we return `null`?

If we drop `ReflectionType` from `getType()` (large union), we also get rid of the assertion below.